arm/Microsoft.ContainerService/managedClusters/agentPools/deploy.bicep

```diff
@@ -0,0 +1,23 @@
+@description('Required. Name of the managed cluster')
+@minLength(1)
+param managedClusterName string
+
+@description('Required. Name of the agent pool')
+param name string
+
+@description('Required. Properties for the container service agent pool profile.')
+param agentPoolProperties object
+
+resource agentPool 'Microsoft.ContainerService/managedClusters/agentPools@2021-05-01' = {
+  name: '${managedClusterName}/${name}'
+  properties: agentPoolProperties
```

Comment:
@lsnoddy Could you please open an issue for this? We should expand this properties object to it's properties to enable parameter validation.
However, I don't won't to overload you as this was not in scope for this PR.

Reply:
Created issue https://github.com/Azure/ResourceModules/issues/497 and assigned myself as I'm familiar and can begin working on this for my next issue.